src/Controller/PageController.php

```diff
@@ -275,4 +276,49 @@ public function healthcheck()
 
         return new JsonResponse(['message' => 'OK']);
     }
+
+    public function statuscheck()
+    {
+        $client = HttpClient::create(['verify_peer' => false, 'verify_host' => false, 'timeout' => 5]);
+        $listOfUrlToCheck = $this->getHeaderAndFooterListFromCMS($client, getenv('APP_CMS_BASE_URL'));
+        $results = [];
+
+        foreach ($listOfUrlToCheck as $url) {
+            try {
+                $response = $client->request('GET', $url);
+                $results[$url] = $response->getstatuscode();
+            } catch (TransportExceptionInterface $e) {
+            }
+        }
+        return $this->render('pages/status_check.html.twig', [
+            'results'           => $results,
+            'totalLinks'        => count($listOfUrlToCheck)
+         ]);
+    }
+
+
+    public function getHeaderAndFooterListFromCMS($client, $APP_CMS_BASE_URL)
+    {
+
+        $numbers = ['21','22','23','24','25'];
+        $returnList = [];
+
+        foreach ($numbers as $number) {
+            $apiUrl = $APP_CMS_BASE_URL . '/wp-json/wp-api-menus/v2/menus/' . $number;
+            $CMSresponse = $client->request('GET', $apiUrl);
+
+            if ($CMSresponse->getStatusCode() == 200) {
+                $jsonObjects = json_decode($CMSresponse->getContent())->items;
+
+                foreach ($jsonObjects as $jsonObject) {
+                    $url = $jsonObject->url;
+
+                    if (substr($url, 0, 4) == 'http') {
```

Comment:
I'm guessing this check is here just in case one or more URL's don't follow HTTPS protocol? Is that correct?

Reply:
This is just to make sure the url is a link, because the Helpline: 0345 410 2222 is also returned from the cms header. So we want to get rid of that since it's not a hyperlink.

Reply:
Ah ok, yep sorry that's what I meant. Thanks.

Reply:
Sorry, do you think this might better to validate if it's a URL? https://www.w3schools.com/php/filter_validate_url.asp

Reply:
I will look into this, thanks for the tips